lib/matplotlib/axes/_axes.py

```diff
@@ -3369,22 +3375,27 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         everymask = np.zeros(len(x), bool)
         everymask[offset::errorevery] = True
 
-        def xywhere(xs, ys, mask):
-            """
-            Return xs[mask], ys[mask] where mask is True but xs and
-            ys are not arrays.
-            """
-            assert len(xs) == len(ys)
-            assert len(xs) == len(mask)
-            xs = [thisx for thisx, b in zip(xs, mask) if b]
-            ys = [thisy for thisy, b in zip(ys, mask) if b]
-            return xs, ys
+        def apply_mask(arrays, mask):
+            # Return, for each array in *arrays*, the elements for which *mask*
+            # is True, without using fancy indexing.
+            return [[*itertools.compress(array, mask)] for array in arrays]
 
-        def extract_err(name, err, data):
+        def extract_err(name, err, data, lolims, uplims):
             """
-            Private function to parse *err* and subtract/add it to *data*.
-
-            Both *err* and *data* are already iterables at this point.
+            Private function to compute error bars.
+
+            Parameters
+            ----------
+            name : {'x', 'y'}
+                Name used in the error message.
+            err : iterable
```

Comment:
```suggestion
            err : array-like
```
also for the other parameters in the docstrin.